util/src/main/resources/org/killbill/billing/util/customfield/dao/CustomFieldSqlDao.sql.stg

```diff
@@ -67,3 +67,33 @@ searchQuery(prefix) ::= <<
   or <prefix>field_name like :likeSearchKey
   or <prefix>field_value like :likeSearchKey
 >>
+
+getSearchCountByObjectTypeAndFieldName(fieldValue) ::= <<
+select
+count(1) as count
+from <tableName()>
+where
+object_type = :objectType
```

Comment:
Do we need new indices or are we all set?

Reply:
With our current indexes, we are relying on the existing `custom_fields_tenant_account_record_id`:

```
+----+-------------+---------------+------------+------+-----------------------------------------------------+----------------------------------------+---------+-------+------+----------+----------------------------------------------------+
| id | select_type | table         | partitions | type | possible_keys                                       | key                                    | key_len | ref   | rows | filtered | Extra                                              |
+----+-------------+---------------+------------+------+-----------------------------------------------------+----------------------------------------+---------+-------+------+----------+----------------------------------------------------+
|  1 | SIMPLE      | custom_fields | NULL       | ref  | custom_fields_tenant_account_record_id,my_new_index | custom_fields_tenant_account_record_id | 8       | const |    1 |    50.00 | Using index condition; Using where; Using filesort |
+----+-------------+---------------+------------+------+-----------------------------------------------------+----------------------------------------+---------+-------+------+----------+----------------------------------------------------+

```

I tried the following indexes:
* Composite (`tenant_record_id`, `field_name`)
* Composite (`field_name`, `tenant_record_id`)
* Simple (`field_name`)

None of those get used and mysql keeps relying on existing `(tenant_record_id, account_record_id);`, which is not ideal...


If i force the index, then i see mysql using it, but it worth it -- and how portable is this for other engines?

Reply:
Wondering if it's because you don't have enough data in the table (and any one of them is good enough)?

Let's keep an eye on it in production and adjust as data grows.

Reply:
Yep, agreed... we could retry on G. DB as well to see what we see...